pac4j-core/src/main/java/org/pac4j/core/context/J2EContext.java

```diff
@@ -136,7 +136,7 @@ public void writeResponseContent(final String content) {
 
     @Override
     public void setResponseStatus(final int code) {
-        if (code == HttpConstants.OK || code == HttpConstants.TEMP_REDIRECT) {
+        if (code < 400) {
```

Comment:
No, because I'm using 400 as a threshold, not to test the `BAD_REQUEST` condition specifically.

A condition like `code < BAD_REQUEST` would negate the purpose of using a constant, because the logic only makes sense if you know that `BAD_REQUEST` is `400` and thus the very first possible HTTP error status value.

If you really don't want the 400 right here, the right way to do this would be to make HttpConstants a true enum with a method like `isError` and maybe even distinguish `isRequestError` (4xx response) from `isServerError` (5xx response).   That would be a good refactor if this logic pops up in more than one place.